.github/workflows/go-getter.yml

```
@@ -0,0 +1,156 @@
+name: go-getter
+
+on: [push]
+
+env:
+  TEST_RESULTS_PATH: /tmp/test-results
+
+jobs:
+
+  linux-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version:
+          - 1.19
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+      
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Create test directory
+        run: |
+          mkdir -p ${{ env.TEST_RESULTS_PATH }}
+
+      - name: Download go modules
+        run: go mod download
+      
+      - name: Cache / restore go modules
```

Optional consistency nit, different name from the other step name for Windows tests:

```suggestion
      - name: Setup cache for go modules
```